examples/generative/bayesian_neural_networks_wine.py

```diff
@@ -0,0 +1,423 @@
+We use the red wine subset, which contains 4,898 examples.
+The dataset has 11numerical physicochemical features of the wine, and the task
+is to predict the wine quality, which is a score between 0 and 10.
+In this example, we treat this as a regression task.
+
+You can install TensorFlow Datasets using the following command:
+
+```python
+pip install tensorflow-datasets
+```
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow_datasets as tfds
+import tensorflow_probability as tfp
+
+"""
+## Create training and evaluation datasets
+
+Here, we load the `wine_quality` dataset using `tfds.load()`, and we convert
+the target feature to float. Then, we shuffle the dataset and split it into
+training and test sets. We take the first `train_size` examples as the train
+split, and the rest as the test split.
+"""
+
+
+def get_train_and_test_splits(train_size, batch_size=1):
+    dataset = (
+        tfds.load(name="wine_quality", as_supervised=True, split="train")
+        .shuffle(buffer_size=dataset_size)
+        .map(lambda x, y: (x, tf.cast(y, tf.float32)))
+    )
+
+    train_dataset = (
+        dataset.take(train_size)
+        .shuffle(buffer_size=train_size)
+        .batch(batch_size)
+        .repeat(1)
+    )
+    test_dataset = dataset.skip(train_size).batch(batch_size).repeat(1)
+
+    return train_dataset, test_dataset
+
+
+"""
+## Compile, train, and evaluate the model
+"""
+
+hidden_units = [8, 8]
+learning_rate = 0.001
+
+
+def run_experiment(model, loss, train_dataset, test_dataset):
+
+    model.compile(
+        optimizer=keras.optimizers.RMSprop(learning_rate=learning_rate),
+        loss=loss,
+        metrics=[keras.metrics.RootMeanSquaredError()],
+    )
+
+    print("Start training the model...")
+    model.fit(train_dataset, epochs=num_epochs)
+    print("Model training finished.")
+    _, rmse = model.evaluate(train_dataset, verbose=0)
+    print(f"Train RMSE: {round(rmse, 3)}")
+
+    print("Evaluating model performance...")
+    _, rmse = model.evaluate(test_dataset, verbose=0)
+    print(f"Test RMSE: {round(rmse, 3)}")
+
+
+"""
+## Create model inputs
+"""
+
+FEATURE_NAMES = [
+    "fixed acidity",
+    "volatile acidity",
+    "citric acid",
+    "residual sugar",
+    "chlorides",
+    "free sulfur dioxide",
+    "total sulfur dioxide",
+    "density",
+    "pH",
+    "sulphates",
+    "alcohol",
+]
+
+
+def create_model_inputs():
+    inputs = {}
+    for feature_name in FEATURE_NAMES:
+        inputs[feature_name] = layers.Input(
+            name=feature_name, shape=(1,), dtype=tf.float32
+        )
+    return inputs
+
+
+"""
+## Experiment 1: standard neural network
+
+We create a standard deterministic neural network model as a baseline.
+"""
+
+
+def create_baseline_model():
+    inputs = create_model_inputs()
+    features = keras.layers.concatenate(list(inputs.values()))
```

Comment:
You should sort the features by name to avoid indeterminism

Reply:
Done